### PR TITLE
fix(balance): improved consensus on getbalance

### LINKED ISSUE #523: Consensus on getBalance
When getting balance we check only Value of balance for consensus, but we must check  whole json returned for the request, probably hashing all the fields in defined order and comparing them

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed https://github.com/0chain/gosdk/issues/523

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
-

### CODE REVIEW

File: zcncore/http_consensus_maps.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	"errors"
+	"net/http"
+	"sort"
+	"strings"
+)
+
+var ErrNilHttpConsensusMaps = errors.New("nil_httpconsensusmaps")
+
+type HttpConsensusMaps struct {
+	ConsensusThresh float32
```
Comment (boddumanohar) [inline zcncore/http_consensus_maps.go]: why use `float32`? Recently we have migrated consensous `ConsensusThresh` to use it type.

File: zcncore/http_consensus_maps.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	}
+
+	m, hash, err := c.buildMap(respBody)
+	if err != nil {
+		return err
+	}
+
+	consensus := c.WinMapConsensus[hash]
+	consensus++
+	c.WinMapConsensus[hash] = consensus
```
Comment (dabasov) [inline zcncore/http_consensus_maps.go]: it is possible to simply
```go
c.WinMapConsensus[hash]++
```

File: zcncore/wallet_base.go [hunk 3]
```diff
@@ -636,44 +636,45 @@ func getBalanceFieldFromSharders(clientID, name string) (int64, string, error) {
...
-	var winInfo string
-	var winError string
+
+	consensusMaps := NewHttpConsensusMaps(consensusThresh)
+
 	for i := 0; i < numSharders; i++ {
 		rsp := <-result
+
 		logging.Debug(rsp.Url, rsp.Status)
 		if rsp.StatusCode != http.StatusOK {
```
Comment (dabasov) [inline zcncore/wallet_base.go:646]: looks like this body is the same as out of if, do we really need this unsuccessful result case?

**Review by dabasov — APPROVED** [review]

File: zcncore/http_consensus_maps.go (hunk 1 above)
Comment (cnlangzi) [reply zcncore/http_consensus_maps.go]: fixed

File: zcncore/http_consensus_maps.go (hunk 2 above)
Comment (cnlangzi) [reply zcncore/http_consensus_maps.go]: fixed

File: zcncore/wallet_base.go (hunk 3 above)
Comment (cnlangzi) [reply zcncore/wallet_base.go:646]: fixed

**Pull request merged**